Classes/Service/ImportService.php

```diff
@@ -0,0 +1,180 @@
+        $dataHandler->start($importJob->getData(), $cmd);
+        $dataHandler->process_cmdmap();
+        $dataHandler->process_datamap();
+    }
+
+    /**
+     * @param array<string, array<int|string, array<string, mixed>>> $insertData
+     * @param string $table
+     * @return array<string, array<int|string, array<string, mixed>>>
+     * @deprecated Using Hooks is deprecated since 5.0. Use Events instead
+     */
+    private function callHook(array $insertData, string $table): array
+    {
+        if (count($GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['SUDHAUS7\\Xlsimport\\Controller\\XlsimportController']['Hooks'] ?? []) > 0) {
+            trigger_deprecation(
+                'xlsimport',
+                '5.0',
+                sprintf(
+                    'Using hooks for manipulating relations is deprecated and will be removed with version 6. Use "%s" instead.',
+                    ManipulateRelationsEvent::class
+                ),
+            );
+            foreach ($GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['SUDHAUS7\\Xlsimport\\Controller\\XlsimportController']['Hooks'] as $_classRef) {
+                $hookObj = GeneralUtility::makeInstance($_classRef);
+                if (method_exists($hookObj, 'manipulateRelations')) {
+                    $hookObj->manipulateRelations($insertData, $table, $this);
+                }
+            }
+        }
+        return $insertData;
+    }
+
+    /**
+     * @throws FileDoesNotExistException
+     * @throws \JsonException
+     * @return array<array-key, mixed>
+     */
+    private function loadDataFromJsonFile(string $jsonFileName): array
+    {
+        $jsonFile = $this->buildJsonFileName($jsonFileName);
+        if (!is_file($jsonFile)) {
+            throw new FileDoesNotExistException(
+                'The JSON file saved temporarily was not found',
+                1705162066162
+            );
+        }
+        return json_decode(file_get_contents($jsonFile) ?: '', true, 512, JSON_THROW_ON_ERROR);
+    }
+
+    private function buildJsonFileName(string $jsonFileName): string
+    {
+        $temporaryPath = Environment::getVarPath() . '/transient/';
+        return sprintf('%s%s', $temporaryPath, $jsonFileName);
+    }
+
+    /**
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws DBALException
+     * @return array<string, array<int|string, array<string, int>>>
+     */
+    private function prepareRecordDeleteByJob(ImportJob $job): array
+    {
+        $db = GeneralUtility::makeInstance(ConnectionPool::class);
+        $builder = $db->getQueryBuilderForTable($job->getTable());
+        $ids = $builder->select('uid')->from($job->getTable())->where(
+            $builder->expr()->eq('pid', $job->getPid())
+        )->executeQuery()->fetchAllAssociative();
+
+        $cmd = [];
+        if (is_array($ids)) {
+            foreach ($ids as $id) {
+                $cmd[$job->getTable()][$id['uid']]['delete'] = 1;
```

Comment:
Nice to see, that a DataHandler command map is created for record deletion,
and letting the core determine if soft-delete or real deleate is suitable
for that table. (Possible as we are in a backend context with this service).